NEWS.md

```diff
@@ -54,6 +54,11 @@ Compiler/Runtime improvements
   Code heavily dependent on constant propagation should see significant compile-time performance improvements and
   certain cases (e.g. calls to uninlinable functions that are nevertheless effect free) should see runtime performance
   improvements. Effects may be overwritten manually with the `@Base.assume_effects` macro ([#43852]).
+* Precompilation now saves more type-inferred code, resulting in reduced time-to-first task for many packages.
```

Comment:
Is this only when explicit `precompile`s (or representative workloads) have been used?